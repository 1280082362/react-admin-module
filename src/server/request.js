import axios from "axios";
import config from "./config";
import { message } from "antd";

import cookie from "react-cookies";
import history from "../routers/history";

const requestList = [];
axios.defaults.timeout = 10000;
axios.defaults.withCredentials = true;
axios.defaults.crossDomain = true;
//http request 拦截器
axios.interceptors.request.use(
  config => {
    requestList.push(config.url);
    const token = cookie.load("token");
    if (token) {
      config.headers = {
        "Content-Type": "application/json;charset=UTF-8",
        token: token
      };
    } else {
      config.headers = {
        "Content-Type": "application/json;charset=UTF-8"
      };
    }
    return config;
  },
  error => {
    return Promise.reject(error);
  }
);
//http response 拦截器
axios.interceptors.response.use(
  response => {
    requestList.splice(
      requestList.findIndex(item => item === response.config.url),
      1
    );
    return response;
  },
  err => {
    requestList.length = 0;
    if (err && err.response) {
      switch (err.response.status) {
        case 400:
          err.message = "请求错误";
          break;

        case 401:
          err.message = "请求错误";
          break;

        case 403:
          err.message = "请求错误";
          /*token错误路由重定向*/
          history.push("/login");
          break;

        case 404:
          err.message = "请求错误";
          break;

        case 408:
          err.message = "请求超时";
          break;

        case 500:
          err.message = "请求错误";
          break;

        case 501:
          err.message = "服务未实现";
          break;

        case 502:
          err.message = "网关错误";
          break;

        case 503:
          err.message = "服务不可用";
          break;

        case 504:
          err.message = "网关超时";
          break;

        case 505:
          err.message = "HTTP版本不受支持";
          break;

        default:
      }
    } else {
      err.message = "网络出现问题，请稍后再试";
    }
    message.warning(err.message);
    return Promise.reject(err);
  }
);

/**
 * 封装get方法
 * @param url
 * @param data
 * @returns {Promise}
 */

function get(url, params = {}) {
  return new Promise((resolve, reject) => {
    axios
      .get(config + url, {
        params: params
      })
      .then(response => {
        resolve(response.data);
      })
      .catch(err => {
        reject(err);
      });
  });
}

/**
 * 封装post请求
 * @param url
 * @param data
 * @returns {Promise}
 */

function post(url, data = {}) {
  return new Promise((resolve, reject) => {
    axios.post(config + url, data).then(
      response => {
        resolve(response.data);
      },
      err => {
        reject(err);
      }
    );
  });
}

export default {
  get,
  post
};
