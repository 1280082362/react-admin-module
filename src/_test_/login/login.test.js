import React from "react";
import renderer from "react-test-renderer";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Login from "../../view/login/login";
import request from "../../server/login/login";

configure({ adapter: new Adapter() });

let taskList;

describe("loginModule", () => {
  /*快照测试*/
  it("snapShootTest", () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  /*异步接口测试*/
  it(" asynchronization", async () => {
    expect.assertions(1); //测试接口是否成功及参数是否正确，需要返回的断言数量
    taskList = await request.loginFun(); //请求数据
    expect(taskList.constructor).toBe(Object); //判断数据类型为数组
  });
  /*dom 操作测试*/
  // test('dom操作', () => {
  //   const wrapper =  mount(<HomePage taskList={taskList}/>)        //将获取的数据导入组件
  //   expect(wrapper.find('.item').length).toBe(taskList.length);                 //判断渲染的节点是否正确
  //   expect(wrapper.find('.item').at(0).find('p').at(0).text()).toBe(taskList[0].name)            //判断渲染的节点内容是否正确
  //   expect(Number(wrapper.find('.item').at(0).find('p').at(1).text())).toBe(taskList[0].age)            //判断渲染的节点内容是否正确
  //   expect(wrapper.find('.clicks-0').length).toEqual(1);
  //   wrapper.find('a').simulate('click');                                //模拟点击事件，可传参数
  //   expect(wrapper.find('.clicks-1').length).toEqual(1);
  // })
});
