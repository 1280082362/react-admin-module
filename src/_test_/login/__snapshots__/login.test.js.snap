// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginModule snapShootTest 1`] = `
<div>
  登录
</div>
`;
